influxdb_iox/src/commands/remote/partition.rs

```diff
@@ -49,5 +105,489 @@ pub async fn command(connection: Connection, config: Config) -> Result<(), Error
+                partition.key
+            );
+            let parquet_files = catalog_client
+                .get_parquet_files_by_partition_id(partition_mapping.remote_partition_id)
+                .await?;
+
+            let parquet_files =
+                load_parquet_files(&catalog, schema.id, partition_mapping, parquet_files).await?;
+
+            let mut handles = vec![];
+            let store_client = store::Client::new(connection);
+            for parquet_file in parquet_files {
+                let path = ParquetFilePath::new_new_gen(
+                    parquet_file.namespace_id,
+                    parquet_file.table_id,
+                    parquet_file.sequencer_id,
+                    parquet_file.partition_id,
+                    parquet_file.object_store_id,
+                )
+                .absolute_dirs_and_file_name();
+                let path = object_store.path_from_dirs_and_filename(path);
+                match object_store.get(&path).await {
+                    Ok(_) => {
+                        println!(
+                            "skipping file {} already in the object store",
+                            parquet_file.object_store_id
+                        );
+                    }
+                    Err(object_store::Error::NotFound { .. }) => {
+                        println!("getting file {} from remote", parquet_file.object_store_id);
+                        let object_store = Arc::clone(&object_store);
+                        let mut store_client = store_client.clone();
+                        let task = tokio::task::spawn(async move {
+                            let mut res = store_client
+                                .get_parquet_file_by_object_store_id(
+                                    parquet_file.object_store_id.to_string(),
+                                )
+                                .await
+                                .expect("error getting file from remote");
+                            let mut bytes = Vec::new();
+
+                            while let Some(Ok(next)) = res.next().await {
+                                bytes.extend_from_slice(next.data.as_ref())
+                            }
+                            let bytes = Bytes::from(bytes);
+                            object_store
+                                .put(&path, bytes)
+                                .await
+                                .expect("error putting file in object store");
+                            println!(
+                                "wrote file {} to object store",
+                                parquet_file.object_store_id
+                            );
+                        });
+                        handles.push(task);
+                    }
+                    e => return Err(Error::ObjectStore(e.unwrap_err())),
+                }
+            }
+
+            join_all(handles).await;
```

Comment:
If one of the tokio tasks panics, it is not propagated to the main thread and so the program will still return 0 / OK which can be annoying if scripting!

This code iterates over the task handles and panics if any of the tokio tasks have panicked (due to an `unwrap()` or `expect()`):

```suggestion
            join_all(handles)
                .await
                .into_iter()
                .try_collect::<Vec<_>>()
                .expect("worker thread crashed");
```

With the above, a failed task will return non-0 👍